WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -175,13 +179,22 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
     /// Synchronizes the specified WordPress Account.
     ///
     func sync(credentials: WordPressCredentials, onCompletion: @escaping () -> Void) {
-        guard case let .wpcom(username, authToken, _, _) = credentials else {
+        guard case let .wpcom(authToken, _, _) = credentials else {
             fatalError("Self Hosted sites are not supported. Please review the Authenticator settings!")
         }
 
-        StoresManager.shared
-            .authenticate(credentials: .init(username: username, authToken: authToken))
-            .synchronizeEntities(onCompletion: onCompletion)
+        let tempUsername = UUID().uuidString
+        StoresManager.shared.authenticate(credentials: .init(username: tempUsername, authToken: authToken))
+        let action = AccountAction.synchronizeAccount { (account, error) in
+            if let account = account {
+                StoresManager.shared
+                    .authenticate(credentials: .init(username: account.username, authToken: authToken))
+                    .synchronizeEntities(onCompletion: onCompletion)
+            } else {
+                StoresManager.shared.synchronizeEntities(onCompletion: onCompletion)
+            }
+        }
+        StoresManager.shared.dispatch(action)
```

Comment:
I like your work around here @aerych with the temp username (UUID). 

Initially, I wondered what the ramifications were if the completion block on `AccountAction.synchronizeAccount` comes back with a nil `account` (for whatever strange reason).  So while testing this, I forced that to happen by commenting out some of the code above. E.g.:

```
        let action = AccountAction.synchronizeAccount { (account, error) in
                StoresManager.shared.synchronizeEntities(onCompletion: onCompletion)
        }
```

and the app recovered from any bad state it could have been in with the username being the temp one. 🤷‍♂️

Reply:
Oh good.  My hope was it would sort it self out either on the next launch or the next time it refreshed account details.  I'd like to better understand what happens to the username set in default credentials and confirm that it gets updated and where just to be safe.